quantumarticle.cls

```diff
@@ -222,23 +220,28 @@ class for Quantum - the open journal for quantum science (http://quantum-journal
 % set date to be empty as a default, otherwise latex will fill it with \@today which triggers the allowtoday error
 \date{}
 
-
+\newtoggle{@noarxiv}
+\togglefalse{@noarxiv}
 \DeclareOptionX{noarxiv}
 {
-	\ExecuteOptionsX{a4paper,allowtoday,allowfontchangeintitle,nopdfoutputerror,unpublished}
+  \ExecuteOptionsX{a4paper,allowtoday,allowfontchangeintitle,nopdfoutputerror,unpublished}
+  \toggletrue{@noarxiv}
 }
 
 
 \ExecuteOptionsX{10pt,oneside,twocolumn,notitlepage,final}
 \ProcessOptionsX
-
+% Throw an error if no paper size was manually specified
+% Simply defaulting to a4paper gives inconsistent behaviour when submitting to the arXiv as the arXiv seems to sometimes set a4paper and sometimes letterpaper, maybe depending on the geographic region of the submitter? 
 \iftoggle{@papersizespecified}
 {
 }
 {
 	\setlength\paperheight {297mm}%
-	\setlength\paperwidth  {210mm}
-	\ClassError{quantumarticle}{Please specify a paper size. Available options: a4paper, a5paper, b5paper, letterpaper, legalpaper, executivepaper}{}
+	\setlength\paperwidth  {210mm}%
+        \iftoggle{@noarxiv}{}{%
```

Comment:
This might be superfluous as we will never end up in the else part of the if if `noarxiv` is given. This is because `noarxiv` itself calls the option `a4paper` which in turn sets `@papersizespecified` to `true`.

The expected behaviour is that if `noarxiv` is given then the user should not be forced to specify a papersize, right? In this case one could get rid of the surrounding if-clause (maybe we should keep it though, because `noarxiv` is the thing that disables the 'disciplinary features').

Reply:
I agree that it is in principle superfluous and admit that I was not fully aware of that when I put the if condition into the code. However, I also agree that it is better to leave it there. Who knows, maybe one day we decide that `noarxiv` should not set the `a4paper` option and then we have a bug that will be hard to find.